packages/calcite-components/src/components/accordion-item/accordion-item.scss

```diff
@@ -167,6 +181,10 @@
   inline-size: 100%;
 }
 
+.slot-content-start {
+  margin-inline-end: var(--calcite-internal-accordion-icon-margin);
```

Comment:
We also need to add this for margin-inline-start on content-end to prevent text from running into slotted content.